src/misc/setup.cpp

```diff
@@ -1234,7 +1234,33 @@ bool CommandLine::FindExist(const char *name, bool remove)
 	return true;
 }
 
-bool CommandLine::FindInt(const char *name, int &value, bool remove)
+// Checks if any of the command line arguments are found in the anterior
+// arguments and come before any of the posterior arguments. If none of the
+// command line arguments are found in the anterior then false is returned.
+bool CommandLine::ExistsPriorTo(const std::list<std::string>& anterior,
+                                const std::list<std::string>& posterior) const
+{
+	auto any_of_iequals = [](const std::list<std::string>& haystack,
+	                         const std::string& needle) {
+		return std::any_of(haystack.begin(),
+		                   haystack.end(),
+		                   [&needle](const std::string& hay) {
+			                   return iequals(hay, needle);
+		                   });
+	};
+
+	for (const auto& cmd : cmds) {
+		if (any_of_iequals(anterior, cmd)) {
```

Comment:
`cmd` is not really a "command" but rather an `arg`(ument), isn't it?

Reply:
I agree; but I was following the original design. It starts with the class "CommandLine" which stores all the CLI insides a "cmds" std::list member variable, which this code is iterating over.

I like to use `for (const auto& singular : plural)`, in this case `cmd` and `cmds`.

But yeah.. cmds would be better named as args. 

So I've named the singular `cli_arg`, just to make it more clear that this is the actual arg from the command like, where as the pre and post args are just canned sets for comparison.

So